### PR TITLE
Pool classes no explicit mirror requirement

### PR DESCRIPTION
**Pull request by inkdot7**
Please ignore this as a usual PR.
Used to test a commit that removes the mirror requirement for special pool allocation class.  When working, would be suggested to be squashed into #5182

### CODE REVIEW

**Comment by kpande** [comment]
why would you want an unsafe pool geometry?

**Comment by DeHackEd** [comment]
First because it's useful for testing. Also this should permit RAID-Z based metadata pools.

**Comment by kpande** [comment]
the original idea behind 5182 is that the redundancy of the DMU/MOS/DDT device would have to match that of the pool before the mirror requirement is not enforced.

also, you have changed the logic of 5182 by allowing metadata devices to be split. I am not sure this is a good idea.

**Comment by inkdot7** [comment]
> the original idea behind 5182 is that the redundancy of the DMU/MOS/DDT device would have to match that of the pool before the mirror requirement is not enforced.

Sorry, I see that I somehow managed to circumvent the normal ZFS checks.  This was not the intention, will look into that and fix.  Thanks!

> also, you have changed the logic of 5182 by allowing metadata devices to be split. I am not sure this is a good idea.

Allowing splits would be a logical consequence of not explicitly enforcing mirrors.  I do not see any technical reasons to disallow it?

**Comment by kpande** [comment]
split only occurs for mirrors and allowing it for every level of redundancy could be problematic, is what I meant. please ensure that splitting a stripe is not somehow going to be allowed.

File: module/zfs/spa.c [hunk 1]
```diff
@@ -5487,15 +5478,6 @@ spa_vdev_split_mirror(spa_t *spa, char *newname, nvlist_t *config,
 			}
 		}
 
-		/* disallow splitting for allocation classes */
```
Comment (kpande) [inline module/zfs/spa.c]: please ensure we cannot split non-mirrored vdev.

**Pull request closed without merging**